### DISCUSSION TITLE
Breaking change in version 6.0.6 for EntityDTO?

### DISCUSSION THREAD

**Discussion by sualko** [mikro-orm/mikro-orm#5186] [General]
Hi everyone,

in general I do not fully understand all those entity types like, `EntityData`, `EntityDTO`, or `EntityProps`, and it seams there is no documentation when and how to use them. So maybe my issue with version 6.0.6 originates from my misunderstanding. Now back to my problem: With the latest bug fix release `EntityDTO` returns an union type with undefined for nullable fields. In 6.0.5 this was not the case and I do not understand the change and think it is also breaking. At least in the mysql world, a field can not be undefined (except it was not selected). I know that the documentation recommends to mark all nullable properties as optional, but I do not understand this either.

So why was the undefined added to the `EntityDTO` type? Since I use `toObject` a lot together with some interfaces, this breaks my whole code. 

I found no further explanation in the commit https://github.com/mikro-orm/mikro-orm/commit/1691a799ff762e1ecde15388433cd2544e42fc5e

Here a little example:
```
@Entity()
class Book {
  @PrimaryKey()
  id: number;

  @Property()
  title: string | null;
}

type TBook = EntityDTO<Book>;  // { id: number, title?: string | null | undefined }
```

**Comment by B4nan** [comment]
I can't help you without seeing some reproduction.

**Reply by sualko** [reply]
I added an example above. Here is the type hint in vsc.

![image](https://github.com/mikro-orm/mikro-orm/assets/2974196/b37e7c37-af5b-4770-bb1d-0b7b9cb1a955)

The issue is that the `title` is optional now and undefinable.

**Reply by B4nan** [reply]
Just to set some expectations, your `title` property is nullable, and that is going to stay (in the DTO as well), the only problem I see that needs fixing is the `undefined` type in there, correct?

**Reply by sualko** [reply]
> the only problem I see that needs fixing is the undefined type in there, correct?

The `undefined` and I'm wondering, why the property is optional now. I do not get the point of the whole commit. Can you give me some background information on that? Maybe I understand the whole thing than better.

Here the type on 6.0.5, which is what I would expect.

![image](https://github.com/mikro-orm/mikro-orm/assets/2974196/b3e48b02-c542-47a2-80c8-35b4b969eb06)

**Reply by B4nan** [reply]
The commit was fixing something else, it wasn't supposed to break this. The optionality wasn't properly respected in some cases.

**Reply by B4nan** [reply]
This will help d661c68dd8f529e92fc2275ce72fda0341d4e7bb

**Reply by sualko** [reply]
Thanks @B4nan. That was a quick fix and release. :clap:  :tada:
